packages/debug-adapter/src/debug-session.ts

```diff
@@ -115,33 +99,33 @@ export class WollokDebugSession extends DebugSession {
   }
 
   protected setBreakPointsRequest(response: DebugProtocol.SetBreakpointsResponse, args: DebugProtocol.SetBreakpointsArguments, _request?: DebugProtocol.Request): void {
-    const sourcePath = this.toWollokPath(args.source.path)
-    const breakpointsPackage = this.environment.descendants.find<Package>(function (node): node is Package {
-      return node.is(Package) && node.sourceFileName === sourcePath
-    })
+    const breakpointsPackage = this.packageFromSource(args.source as Source)
 
+    const breakpointsToRemove = []
     this.executionDirector.breakpoints.forEach(breakpointedNode => {
-      if(breakpointedNode.sourceFileName === breakpointsPackage.sourceFileName) {
-        this.executionDirector.removeBreakpoint(breakpointedNode)
+      if(breakpointedNode.parentPackage.id === breakpointsPackage.id) {
+        breakpointsToRemove.push(breakpointedNode)
       }
     })
-
-    if(breakpointsPackage){
-      const sentences = breakpointsPackage.descendants.filter<Sentence>(function (node): node is Sentence {
-        return node.is(Sentence) && node.parent.is(Body) && args.breakpoints.map(breakpoint => breakpoint.line).includes(node.sourceMap?.start.line)
+    breakpointsToRemove.forEach(breapointedNode => this.executionDirector.removeBreakpoint(breapointedNode))
+
+    if(breakpointsPackage) {
+      const nodesToBreakAt = breakpointsPackage.descendants.filter((node: Node) => {
+        return args.breakpoints.some(breakpoint =>
+          breakpoint.column ?
+            this.positionConverter.convertDebuggerLineToClient(node.sourceMap?.start.line) === breakpoint.line && this.positionConverter.convertDebuggerColumnToClient(node.sourceMap?.start.column) === breakpoint.column :
+            node.is(Sentence) && node.parent.is(Body) && this.positionConverter.convertDebuggerLineToClient(node.sourceMap?.start.line) === breakpoint.line
+        )
```

Comment:
Acá no se debería reutilizar una lógica como la de `cursorNode`? Que vaya de Location a Node.

Reply:
Si, el tema es que esa logica esta en otro package (server), en su momento queria tener un package aparte que sea de logica compartida entre server, client y debugger. PERO el tema es que entre ellos conflictuan los tipados para las mismas entidades. Una position en el server no es lo mismo que una en el debugger.

Reply:
Sí, habría que pensar una API más piola que se pueda reutilizar en to'lado